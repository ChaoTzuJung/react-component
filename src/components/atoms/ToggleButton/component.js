import React from 'react';

import style from './index.css';

const ToggleButton = ({ toggle, handleClick, closeTitle = 'Close', openTitle = 'Open' }) => (
	<button
		type="button"
		className={toggle ? style.button : style.buttonReverse}
		onClick={handleClick}
	>
		{toggle ? closeTitle : openTitle}
	</button>
);

export default ToggleButton;
