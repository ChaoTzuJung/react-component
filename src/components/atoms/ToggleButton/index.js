import { contain } from 'util/container';

import ToggleButton from './component';

const initState = () => ({
	toggle: true,
});

const mapSetStateToProps = ({ toggle }, { onOpen, onClose, openTitle, closeTitle }, setState) => ({
	toggle,
	handleClick() {
		if (toggle) {
			onClose();
		} else {
			onOpen();
		}
		setState({ toggle: !toggle });
	},
	openTitle,
	closeTitle,
});

export default contain(initState, mapSetStateToProps)(ToggleButton);
