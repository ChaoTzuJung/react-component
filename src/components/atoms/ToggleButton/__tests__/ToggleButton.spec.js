import React from 'react';
import { mount } from 'enzyme';

import ToggleButton from 'components/atoms/ToggleButton';

test('should have default value: Close', () => {
	const component = mount(<ToggleButton onOpen={() => {}} onClose={() => {}} />);
	expect(component.text()).toContain('Close');
});

// Create custom snapshot testing
test('should become Open after click', () => {
	const component = mount(<ToggleButton onOpen={() => {}} onClose={() => {}} />);

	expect(component).toMatchSnapshot();

	// manually trigger the callback
	component.find('button').simulate('click');

	expect(component.text()).toContain('Open');
	// re-rendering would become Open
	expect(component).toMatchSnapshot();

	// manually trigger the callback
	component.find('button').simulate('click');

	expect(component.text()).toContain('Close');
	// re-rendering would become Close
	expect(component).toMatchSnapshot();
});
