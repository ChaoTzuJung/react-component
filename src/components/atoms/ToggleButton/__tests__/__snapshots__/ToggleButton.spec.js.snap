// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should become Open after click 1`] = `
<Contain(ToggleButton)
  onClose={[Function]}
  onOpen={[Function]}
>
  <ToggleButton
    handleClick={[Function]}
    toggle={true}
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </ToggleButton>
</Contain(ToggleButton)>
`;

exports[`should become Open after click 2`] = `
<Contain(ToggleButton)
  onClose={[Function]}
  onOpen={[Function]}
>
  <ToggleButton
    handleClick={[Function]}
    toggle={false}
  >
    <button
      className="buttonReverse"
      onClick={[Function]}
      type="button"
    >
      Open
    </button>
  </ToggleButton>
</Contain(ToggleButton)>
`;

exports[`should become Open after click 3`] = `
<Contain(ToggleButton)
  onClose={[Function]}
  onOpen={[Function]}
>
  <ToggleButton
    handleClick={[Function]}
    toggle={true}
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </ToggleButton>
</Contain(ToggleButton)>
`;
